projecttacoma/bulk-submit-demo
# Data Requirements Visualization Panel

## Summary
Added a panel that displays the JSON content of a $data-requirements request on the selected measure. Also added recoil as a global state manager.

### New Behavior
 - Data Requirements panel is now visible and updates each time a new measure is selected
 - Recoil is used by components to manage globally accessed and updated variables

### Code Changes
 - Added component DataRequirementsViewer.ts
 - Set up recoil and updated all necessary components and tests

## Testing Guidance
 - Could not get unit testing to work for this component. Happy to talk through possible routes forward if anyone has ideas
 - Run `npm test` to check that the existing tests work with recoil changes
 - Pop open the test server and give it a `npm run db:reset` then `npm run connectathon-upload` then `npm start`
 - Run an `npm install` in the bulk submit demo to grab the new dependencies
 - Either create or update your `env.local` file to include `NEXT_PUBLIC_DEQM_SERVER=http://localhost:3000/4_0_1`
 - Start up the demo with `npm run dev`
 - Navigate to `http://localhost:3002`
 - Note the new window and ensure that it says "No Data"
 - Select a measure from the measure dropdown
 - Note that the Data Requirements panel briefly switches to a loading spinner before populating with an interactive JSON viewer
 - Select different measures and ensure the proper data requirements load
 - Check out the soon to be populated `Filters` tab as well.
 - Please LMK if you have any stylistic suggestions. I'm not thrilled with how it looks rn.
